src/architecture/web_api.md

```diff
@@ -0,0 +1,57 @@
+   - The trait methods should each be documented with a link to their definition in the interface.
+   - [Example result](https://github.com/servo/servo/pull/34844/commits/5c842527d89f9c715f27427913a8d5fc6b18d4c7).
+5. Back in the spec, 
+   - look for definitions of futher members for your struct. 
+   - These are usually referred to as "internal slots"([example](https://streams.spec.whatwg.org/#default-reader-internal-slots)), 
+   or as something that is "associated" with the interface([example](https://html.spec.whatwg.org/#message-channels)). 
+6. Read-up on [DOM interfaces](https://github.com/servo/servo/blob/4a5ff01e060293721d10289ec56dbd4c58a0969e/components/script/dom/mod.rs).
+7. Using what you know, for each internal slot of the interface:
+   - Add an appropriate member to the struct(s) added at 4.
+   - If this requires defining other structs or enums, these should derive `JSTraceable` and `MallocSizeOf`([example](https://github.com/servo/servo/pull/34844/commits/7d73370b0c41a1b00f4b25b7e1b8bf9b67430708#diff-2e7f6e100fdbd73318de2dda9b3d3883700be9ebfd028d4412a207e93cb02892R53)).
+   - Any `JSTraceable` structs added above that contains members that must be rooted because they are either [JS values](https://github.com/servo/mozjs/blob/87cabf4e9ddf9fafe19713a3d6bc8c5e6105544c/mozjs/src/gc/collections.rs#L94) or [Dom Objects](https://github.com/servo/servo/blob/9887ad369d65eb362db21c778ae5f00aad74db6c/components/script/dom/bindings/root.rs#L5) should be marked with `#[cfg_attr(crown, crown::unrooted_must_root_lint::must_root)]`. [Example](https://github.com/gterzian/servo/blob/b7688fe916d105ae9023cd2429068f16ecba3574/components/script/dom/readablestreamdefaultcontroller.rs#L120), where the lint is needed due to the presence of a `JSVal`.
+   - If such a struct is assigned to a variable, `impl js::gc::Rootable` for the and use `rooted!` to root the variabe([example](https://github.com/servo/servo/pull/34844/commits/94867eec21e06d59c5479bdaa92ef422bc7b21f9)).
+   - All of this can be changed later, so simply use your best judgement at this stage. 
+   - Add methods for [construction](https://github.com/servo/servo/blob/4a5ff01e060293721d10289ec56dbd4c58a0969e/components/script/dom/mod.rs#L91)(not to be confused with a `Constructor` that is part of the Web API). 
+   - [Example result](https://github.com/servo/servo/pull/34844/commits/7d73370b0c41a1b00f4b25b7e1b8bf9b67430708).
+
+### Part 2: Writing a First Draft
+1. For each methods of the bindings trait referred to at 3 in Part 1:
+   - In general, follow the structure of the spec: if a method calls into another named algorithm, implement that named algorithm as a separate private method of your struct, that the trait methods calls into. If you later realize this private method can be used from other structs, make it `pub(crate)`. 
+   - For each algorithm step in the spec:
+       - Copy the prose from the spec, but ommit the step number(these change too often). 
```

Comment:
* `ommit` -> `omit`

* I don't agree with this point and I don't think it reflects the way code is currently being written in servo.

Reply:
I explained my reasoning in more detail on zulip: [#general > Add doc on web api to servo book. @ 💬](https://servo.zulipchat.com/#narrow/channel/263398-general/topic/Add.20doc.20on.20web.20api.20to.20servo.20book.2E/near/508108067)